### PR TITLE
[Feature] Support optimizer time cost trace

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 1]
```diff
@@ -1071,7 +1071,11 @@ private String buildExplainString(ExecPlan execPlan) {
         if (execPlan == null) {
             explainString += "NOT AVAILABLE";
         } else {
-            explainString += execPlan.getExplainString(parsedStmt.getExplainLevel());
+            if (parsedStmt.getExplainLevel().equals(StatementBase.ExplainLevel.OPTIMIZER)) {
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:1074]: a style recommendation, just personal flavour. 

```
if ()
else if {}
else {}
```

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 23 / 46 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/PlannerProfile.java|3|24|12.50%|[66, 71, 120, 196, 197, 199, 204, 208, 209, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 223]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|2|50.00%|[2986]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|3|66.67%|[1075]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/StatementBase.java|1|1|100.00%|[]|

**Review by Youngwb — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/PlannerProfile.java [hunk 2]
```diff
@@ -184,4 +191,35 @@ public void reset() {
...
+        trace.append(print("Parser", getTime("Parser", times), 1));
+        trace.append(print("Analyzer", getTime("Analyzer", times), 1));
+        trace.append(print("Optimizer", getTime("Optimizer", times), 1));
+        trace.append(print("Optimizer.RuleBaseOptimize",
+                getTime("Optimizer.RuleBaseOptimize", times), 2));
+        trace.append(print("Optimizer.CostBaseOptimize",
+                getTime("Optimizer.CostBaseOptimize", times), 2));
+        trace.append(print("Optimizer.PhysicalRewrite",
+                getTime("Optimizer.PhysicalRewrite", times), 2));
+        trace.append(print("ExecPlanBuild", getTime("ExecPlanBuild", times), 1));
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/PlannerProfile.java:221]: Please help me add ”HMS.x“ in customProperties to trace. thanks.

**Comment by stephen-shelby** [comment]
I want to add the "HMS.X" of timers and customProperties to trace after you merge.

**Review by stephen-shelby — APPROVED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Pull request merged**